### PR TITLE
Tables readme update

### LINKED ISSUE #5906: Update Tables Root README
Update the root `README.md` of the Azure SDK for C++ Tables library to follow the template in [this issue.](https://github.com/Azure/azure-sdk-pr/issues/1745)

### PR DESCRIPTION
**Pull request by gearama**
closes #5906 
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Comment by ronniegeraghty** [comment]
Missing these link references at the bottom: 
[azure_sub]: https://azure.microsoft.com/free/
[azure_cli]: https://docs.microsoft.com/cli/azure

**Review by ronniegeraghty — CHANGES_REQUESTED** [review]

File: sdk/tables/azure-data-tables/README.md [hunk 1]
```diff
@@ -4,23 +4,61 @@ Azure Data Tables is a NoSQL data storage service that can be accessed from anyw
...
 * You must have an [Azure subscription][azure_subscription] and either
     * an [Azure Storage account][azure_storage_account] or
     * an [Azure Cosmos Account][azure_cosmos_account].
 
-#### Create account
+If you use the Azure CLI, replace `<your-resource-group-name>` and `<your-<service-name>-name>` with your own, unique names:
+
+```PowerShell
+az login
+az <service-name> create --resource-group <your-resource-group-name> --name <your-<service-name>-name>
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/tables/azure-data-tables/README.md:24]: done

File: sdk/tables/azure-data-tables/README.md [hunk 2]
```diff
@@ -4,23 +4,61 @@ Azure Data Tables is a NoSQL data storage service that can be accessed from anyw
...
+```batch
+vcpkg new --application
+```
 
-#### Create the client
+To install the Azure \<Service-Name> package via vcpkg:
+To add the Azure \<Service-Name> package to your vcpkg enter the following command (We'll also add the Azure Identity library for authentication):
+
+```batch
+vcpkg add port azure-<service-name>-cpp azure-identity-cpp
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/tables/azure-data-tables/README.md:41]: done

File: sdk/tables/azure-data-tables/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/tables/azure-data-tables/README.md:24]: maybe, i don't know how from cli, i only used the portal

File: sdk/tables/azure-data-tables/README.md [hunk 3]
```diff
@@ -248,6 +307,15 @@ The output of the sample contains the error message:
...
 ```
+## Next steps
+
+The following sections provide several code snippets covering some of the most common Table tasks, including:
+
+* [Creating and deleting a table](#creating-and-deleting-a-table "Creating and deleting a table")
+* [Manipulating entities](#manipulating-entities "Manipulating entities")
+* [Table Service Operations](#table-service-operations "Table Service Operations")
+* [Table Transactions Success](#table-transactions-success "Table Transactions Success")
+* [Table Transactions Error](#table-transactions-error "Table Transactions Error")
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/tables/azure-data-tables/README.md:318]: done

File: sdk/tables/azure-data-tables/README.md [hunk 4]
```diff
@@ -33,6 +71,30 @@ az login
 az storage account show -n mystorageaccount -g MyResourceGroup --query "primaryEndpoints.table"
 ```
 
+### Client (Instantiation & List of any Sub-Clients)
+
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/tables/azure-data-tables/README.md:90]: it is , on line below

File: sdk/tables/azure-data-tables/README.md [hunk 5]
```diff
@@ -33,6 +71,30 @@ az login
...
+  static const char* url = "AZURE_SERVICENAME_URL";
+
+  // Use Azure Default Credential wherever possible
+  auto credential = std::make_shared<Azure::Identity::DefaultAzureCredential>();
+
+   // Create client code. Try to avoid using auto on the first use of any type from our libraries that a customer might not be familiar with. 
+   AzureClientType client = clientCtor(url, credential); 
+}
+```
+
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/tables/azure-data-tables/README.md:112]: line 191

File: sdk/tables/azure-data-tables/README.md [hunk 6]
```diff
@@ -33,6 +71,30 @@ az login
...
+int main (){
+  // Add all variables that are needed for creating the client
+  static const char* url = "AZURE_SERVICENAME_URL";
+
+  // Use Azure Default Credential wherever possible
+  auto credential = std::make_shared<Azure::Identity::DefaultAzureCredential>();
+
+   // Create client code. Try to avoid using auto on the first use of any type from our libraries that a customer might not be familiar with. 
+   AzureClientType client = clientCtor(url, credential);
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/tables/azure-data-tables/README.md:109]: it's in the tests , or in the samples. nobody reads these anyway.

File: sdk/tables/azure-data-tables/README.md [hunk 7]
```diff
@@ -33,6 +71,30 @@ az login
...
+
+We'll be using the `DefaultAzureCredential` to authenticate which will pick up the credentials we used when logging in with the Azure CLI earlier. `DefaultAzureCredential` can pick up on a number of Credential types from your environment and is ideal when getting started and developing. Check out our section on [DefaultAzureCredentials](https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/identity/azure-identity#defaultazurecredential) to learn more.
+
+```cpp
+// Add all necessary includes for client creation and samples below
+#include <azure/identity.hpp> 
+#include <azure/<service-name>.hpp
+
+// Add all using statements for client creation and samples below
+using namespace Azure::<Service::Name>;
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/tables/azure-data-tables/README.md:99]: done

File: sdk/tables/azure-data-tables/README.md [hunk 8]
```diff
@@ -33,6 +71,30 @@ az login
 az storage account show -n mystorageaccount -g MyResourceGroup --query "primaryEndpoints.table"
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/tables/azure-data-tables/README.md:86]: i don't see it

File: sdk/tables/azure-data-tables/README.md [hunk 9]
```diff
@@ -33,6 +71,30 @@ az login
...
 ```
 
+### Client (Instantiation & List of any Sub-Clients)
+
+We'll be using the `DefaultAzureCredential` to authenticate which will pick up the credentials we used when logging in with the Azure CLI earlier. `DefaultAzureCredential` can pick up on a number of Credential types from your environment and is ideal when getting started and developing. Check out our section on [DefaultAzureCredentials](https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/identity/azure-identity#defaultazurecredential) to learn more.
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/tables/azure-data-tables/README.md:91]: it works as i have done the transition of the tests myself

**Review by ronniegeraghty — APPROVED** [review]

**Review by ronniegeraghty — APPROVED** [review]

**Pull request merged**